CMOS-78: Add Fluent Bit tooling info (#91)
couchbaselabs/observability: docs/modules/ROOT/pages/deployment-fluentbit.adoc

For full details here, please refer to the official documentation for installation.

For CMOS needs we must do the following:

. Install Fluent Bit.
. Configure Fluent Bit.

Step one can is deploying the binary and providing a startup script to launch it with Couchbase Server in whatever the preferred OS format is.
Refer to the installation instructions from Fluent Bit for this.

Step two can be satisfied by cloning (or otherwise just copying) the link:https://github.com/couchbase/couchbase-fluent-bit/tree/main/conf[configuration] provided by the official Couchbase Fluent Bit image.
The key is to get all the configuration into the expected location for the local Fluent Bit to use, refer to the deployment guide from Fluent Bit for full details for the specific targets.
Whilst this is not officially supported on-premise, it is intended primarily for CAO deployments, it will function correctly on-premise as well.

[source,console]
----
$ git clone --depth 1 https://github.com/couchbase/couchbase-fluent-bit.git
$ cp -R conf/* <config directory for Fluent Bit on this OS>

> ----

Reviewer: Note on the env vars that need to be set - main one that comes to mind for on-prem is `COUCHBASE_LOGS`

Author: I'm just going to link to the FB page I think as this may be version dependent too. I don't want to duplicate stuff that then needs updating.

Reviewer: Yeah, could just link to https://github.com/couchbase/couchbase-fluent-bit#configuration - with a strong note that the defaults need to be explicitly set when deploying to on-prem (since they're set in the Dockerfile)

Author: The linked blog post and example also covers it

== Kubernetes usage

Couchbase Autonomous Operator (CAO) versions 2.2 onwards already support Fluent Bit log forwarding.
Refer to the link:https://docs.couchbase.com/operator/current/concept-couchbase-logging.html#log-forwarding[CAO documentation] for the full details.

The main configuration required is providing the CMOS Loki endpoint as part of the link:https://docs.couchbase.com/operator/current/resource/couchbasecluster.html#couchbaseclusters-spec-servers-pod[Couchbase cluster configuration].
To do this, use Couchbase Fluent Bit link:https://github.com/couchbase/couchbase-fluent-bit#releases[versions 1.1.2 or greater] and supply the following annotations with appropriate values.

[source,yaml]
----
            pod:
                metadata:
                    annotations:
                        # Match all logs
                        fluentbit.couchbase.com/loki_match: "*"
                        # Send to this SVC
                        fluentbit.couchbase.com/loki_host: loki.default
----

A full example of integrating is available as part of the xref:tutorial-kubernetes.adoc[Kubernetes example].

== On-premise usage

> Fluent Bit link:https://docs.fluentbit.io/manual/installation/supported-platforms[already provides various supported targets] so is straightforward to install on-premise.

Reviewer 2: ```suggestion
Fluent Bit link:https://docs.fluentbit.io/manual/installation/supported-platforms[already provides various supported targets] so on-premise installation is straight forward.
```

> For full details here, please refer to the official documentation for installation.

Reviewer 2: ```suggestion
For more details, please refer to the official documentation(add a link here) for installation.
```

For CMOS needs we must do the following:

. Install Fluent Bit.
. Configure Fluent Bit.

> Step one can is deploying the binary and providing a startup script to launch it with Couchbase Server in whatever the preferred OS format is.

Reviewer 2: ```suggestion
Step one is deploying fluent bit binary with Couchbase Server using a startup script.
```

Refer to the installation instructions from Fluent Bit for this.

> Step two can be satisfied by cloning (or otherwise just copying) the link:https://github.com/couchbase/couchbase-fluent-bit/tree/main/conf[configuration] provided by the official Couchbase Fluent Bit image.

Reviewer 2: ```suggestion
Step two is configuring fluent bit with https://github.com/couchbase/couchbase-fluent-bit/tree/main/conf[configuration] provided by the official Couchbase Fluent Bit image. You can either clone the repo or just copy over the configuration file.
```
